SUS-4321 | WatchedItem::addWatch - remove IGNORE flag from INSERT queries

macbre opened the pull request:
https://wikia-inc.atlassian.net/browse/SUS-4321

```sql
mysql@geo-db-f-master.query.consul[plnordycka]>insert into watchlist(wl_user, wl_namespace, wl_title, wl_notificationtimestamp) values (119245, 3, 'Macbre/rss', NULL);
ERROR 1062 (23000): Duplicate entry '119245-3-Macbre/rss' for key 'wl_user'

--

mysql@geo-db-f-master.query.consul[plnordycka]>insert ignore into watchlist(wl_user, wl_namespace, wl_title, wl_notificationtimestamp) values (119245, 3, 'Macbre/rss', NULL);
Query OK, 0 rows affected, 1 warning (0.00 sec)

mysql@geo-db-f-master.query.consul[plnordycka]>show warnings;
+---------+------+---------------------------------------------------------+
| Level   | Code | Message                                                 |
+---------+------+---------------------------------------------------------+
| Warning | 1062 | Duplicate entry '119245-3-Macbre/rss' for key 'wl_user' |
+---------+------+---------------------------------------------------------+
1 row in set (0.00 sec)
```

Let's check how many DB errors will be logged from these `INSERT` queries. Then - if needed - we can add a check whether it makes sense to perform an `INSERT` query.


### Queries

```sql
> User::newFromName('Macbre')->addWatch( Title::newFromText('2000') )
...
Query plpoznan (DB user: wikia_maint) (4) (master): INSERT /* WatchedItem::addWatch CommandLineInc - ee3eb1ce-09cf-4ba1-90c5-be239e196a43 */ INTO `watchlist` (wl_user,wl_namespace,wl_title,wl_notificationtimestamp) VALUES ('119245','0','2000',NULL),('119245','1','2000',NULL) ON DUPLICATE KEY UPDATE wl_notificationtimestamp = VALUES(wl_notificationtimestamp)

--

mysql@geo-db-dev-db-slave.query.consul[plpoznan]>select * from watchlist where wl_user = 119245 and wl_title = '2000';
+---------+--------------+----------+--------------------------+-------------------------+
| wl_user | wl_namespace | wl_title | wl_notificationtimestamp | wl_wikia_addedtimestamp |
+---------+--------------+----------+--------------------------+-------------------------+
|  119245 |            0 | 2000     | NULL                     | 2018-03-23 08:31:52     |
|  119245 |            1 | 2000     | NULL                     | 2018-03-23 08:31:52     |
+---------+--------------+----------+--------------------------+-------------------------+
2 rows in set (0.00 sec)
```

macbre committed:
SUS-4321 | WatchedItem::addWatch - remove IGNORE flag from INSERT queries

sqreek commented on includes/WatchedItem.php:
```diff
@@ -107,8 +105,15 @@ public function addWatch() {
+		try {
+			// SUS-4321 | This **used** INSERT IGNORE to avoid overwriting the notification
+			// timestamp if there's already an entry for this page
+			$dbw->insert( 'watchlist', $rows, __METHOD__ );
```
maybe `ON DUPLICATE KEY UPDATE id=id`?

macbre replied:
Used `Database::upsert` in daec89c

macbre committed:
SUS-4321 | WatchedItem - use upsert - INSERT INTO + ON DUPLICATE KEY UPDATE

insert into watchlist(wl_user, wl_namespace, wl_title, wl_notificationtimestamp) values (119245, 3, 'Macbre/rss', NULL) ON DUPLICATE KEY UPDATE wl_notificationtimestamp = VALUES(wl_notificationtimestamp);

macbre committed:
Merge branch 'dev' into SUS-4321-watchlist-insert-ignore
